samples/client/petstore/typescript-angular2/default/api/pet.service.ts

```diff
@@ -117,7 +118,7 @@ export class PetService {
                 if (response.status === 204) {
                     return undefined;
                 } else {
-                    return response.json() || {};
+                    return ObjectSerializer.deserialize(response.json(), "Array&lt;Pet&gt;") || {};
```

Comment:
This seems to be related to {{returnType}} in line 99 in api.service.mustache.This is also an issue for the ts-node code generator. 
Is the returnType encoded by the program generating the input for mustache?